umbraco/Umbraco-CMS
# Added necesary classes to define the StringPropertyEditor

### Reason
Frontend package devs need a propertyEditor that stores the value as a string without any configuration for, amongst other things, rapid prototyping.

### Description
- Added a NoopStringValueTypeConfiguration that only defines the non overridable peristence data type
- Added a NoopStringConfigurationEditor that wraps the above configuration to return empty values, as none overridable configuration is possible
- Added a StringPropertyEditor that uses the above configuration and editor to satisfy the " Configuration" requirements to be a DataEditor without actually having configuration.

### Testing

You should be able to 
- Create DataType by Posting to {{baseUrl}}/umbraco/management/api/v1/data-type with
```json
{
  "name": "no config string",
  "propertyEditorAlias": "Umbraco.String",
  "propertyEditorUiAlias": "Umbraco.String.Ui",
  "values": [],
  "id": "2EF54F6A-715A-4705-A2FF-F8FC130D9A01",
  "parentId": null
}
```
- Get the list of datatypes ({{baseUrl}}/umbraco/management/api/v1/tree/data-type/root) and find the datatype created in the previous step
- Create doctype with new datatype by Posting to {{baseUrl}}/umbraco/management/api/v1/document-type with
```json
{
  "alias": "testDocType",
  "name": "testDocType",
  "description": "testDocType",
  "icon": "umb:wand",
  "allowedAsRoot": true,
  "variesByCulture": false,
  "variesBySegment": false,
  "isElement": false,
  "properties": [
    {
      "id": "7A9B8D0F-EC59-440F-A47A-79C9C4E65D25",
      "sortOrder": 0,
      "alias": "testProperty",
      "name": "testProperty",
      "description": "testProperty",
      "dataTypeId": "2EF54F6A-715A-4705-A2FF-F8FC130D9A01",
      "variesByCulture": false,
      "variesBySegment": false
    }
  ]
}
```
- Get the list of documentypes ({{baseUrl}}/umbraco/management/api/v1/tree/document-type/root) and find the doctype you created in the previous step, note down its ID
- Create a new document by Posting {{baseUrl}}/umbraco/management/api/v1/document to with
```json
{
  "values": [
    {
      "culture": null,
      "segment": null,
      "alias": "testProperty",
      "value": "testValue"
    }
  ],
  "variants": [
    {
      "culture": null,
      "segment": null,
      "name": "testDocument"
    }
  ],
  "parentId": null,
  "contentTypeId": "ENTER_DOCTYPE_ID_HERE"
}
```
- Get the list of documents ({{baseUrl}}/umbraco/management/api/v1/tree/document/root) and find the id of the "testDocument" you just created
- Read the property from the saved document by getting it from
{{baseUrl}}/umbraco/management/api/v1/document/THE_ID_YOU_FOUND_IN_PREVIOUS_STEP
